react/react1/week3/react1week3-shifts/src/components/InputShift.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { ViewShiftList } from './ViewShiftList';
+import { SearchShift } from './SearchShift';
+
+import { DataDefaultShiftApi } from './DataDefaultShiftApi';
+
+export function InputShift() {
+  const [inputShift, setInputShift] = useState([]);
+
+  const [name, setName] = useState('');
+  const [start, setStartTime] = useState('');
+  const [end, setEndTime] = useState('');
+  const [searchName, setSearchName] = useState('');
+  const [dataFechingState, setDataFechingState] = useState(false);
+
+  function addShift() {
+    const startEntry = start.split(':');
+    const startMinutes = Number(startEntry[0]) * 60 + Number(startEntry[1]);
+    const endEntry = end.split(':');
+    const endMinutes = Number(endEntry[0]) * 60 + Number(endEntry[1]);
+    const totalMinutes = endMinutes - startMinutes;
+    const hours = Math.floor(totalMinutes / 60);
+    const minutes = totalMinutes % 60;
+
+    if (hours > 0 || minutes > 0) {
+      const addedShift = [...inputShift, { name, start, end }];
+      setInputShift(addedShift);
+      alert('shift saved');
+    } else {
+      setInputShift(inputShift);
```

Comment:
This line is not useful here. Doing something unnecessary. You can remove it.

Reply:
yes, removed line 30